src/components/molecules/UiInteractiveSvg/_internal/UiInteractiveSvgElement.vue

```diff
@@ -24,34 +24,35 @@ import {
   ref,
 } from 'vue';
 import type { Ref } from 'vue';
-import type { AttrsFunc } from '../UiInteractiveSvg.vue';
+import type { InteractiveSvgAttrsFunc } from '../UiInteractiveSvg.vue';
+import type { DefineAttrsProps } from '../../../../types';
 
-defineProps({
+export interface InteractiveSvgElementProps {
   /**
-   *  Use this props to set component tag.
-   */
-  tag: {
-    type: String,
-    default: 'path',
-  },
-});
-const setAttrs: AttrsFunc | Ref<AttrsFunc> | undefined = inject('elementsAttrs');
-const elementsAttrs = (attrs: Record<string, unknown>): Record<string, unknown> => {
-  const elementsAttrs = (setAttrs as Ref<AttrsFunc>).value || setAttrs;
+  *  Use this props to set component tag.
+  */
+  tag?: string;
+}
+export type InteractiveSvgElementAttrsProps = DefineAttrsProps<InteractiveSvgElementProps>;
+
+withDefaults(defineProps<InteractiveSvgElementProps>(), { tag: 'path' });
+const setAttrs: InteractiveSvgAttrsFunc | Ref<InteractiveSvgAttrsFunc> | undefined = inject('elementsAttrs');
+const elementsAttrs: InteractiveSvgAttrsFunc = (attrs) => {
+  const elementsAttrs = (setAttrs as Ref<InteractiveSvgAttrsFunc>).value || setAttrs;
```

Comment:
Q: Is this cast necessary? What can we do to get rid of it, to avoid it? 🤔 🙂

Reply:
thanks, we can use `isRef` 😁